### PR TITLE
Fix List decoding so that we propagate errors

### PR DESCRIPTION
**Pull request by jsancio**

### CODE REVIEW

File: cosmos-json/src/main/scala/com/mesosphere/universe/v3/circe/Decoders.scala [hunk 1]
```diff
@@ -53,6 +54,10 @@ object Decoders {
       }
     }
 
+  implicit val decodeListOfTag: Decoder[List[PackageDefinition.Tag]] = {
+    Decoder.decodeCanBuildFrom[PackageDefinition.Tag, List]
```
Comment (cruhland) [inline cosmos-json/src/main/scala/com/mesosphere/universe/v3/circe/Decoders.scala]: Why not implement a generic `decodeList[A]: Decoder[List[A]]`, with the appropriate typeclass constraints on `A`?

Comment (jsancio) [inline cosmos-json/src/main/scala/com/mesosphere/universe/v3/circe/Decoders.scala]: What do you mean?

Comment (jsancio) [inline cosmos-json/src/main/scala/com/mesosphere/universe/v3/circe/Decoders.scala]: FYI, Circe already has a generic `Decoder[List[A]]` for `A`s that implement `Decoder[A]`. That implementation doesn't do failure propagation.

Comment (jsancio) [inline cosmos-json/src/main/scala/com/mesosphere/universe/v3/circe/Decoders.scala]: Never mind. I think I understand what you mean. Change coming up.

Comment (jsancio) [inline cosmos-json/src/main/scala/com/mesosphere/universe/v3/circe/Decoders.scala]: Done.

File: cosmos-json/src/main/scala/com/mesosphere/universe/common/circe/Decoders.scala [hunk 2]
```diff
@@ -18,4 +18,8 @@ object Decoders {
     )
   }
 
+  implicit def decodeListA[A](implicit decoder: Decoder[A]): Decoder[List[A]] = {
+    Decoder.decodeCanBuildFrom[A, List]
+  }
```
Comment (cruhland) [inline cosmos-json/src/main/scala/com/mesosphere/universe/common/circe/Decoders.scala]: Yep, that's what I meant! If this is a stop-gap solution until the fix is made in Circe, could you add a comment and/or file a JIRA so we know to remove this?

**Comment by cruhland** [comment]
LGTM, but see my new comment.

File: cosmos-json/src/main/scala/com/mesosphere/universe/common/circe/Decoders.scala (hunk 2 above)
Comment (jsancio) [inline cosmos-json/src/main/scala/com/mesosphere/universe/common/circe/Decoders.scala]: Done. Added a comment.

**Comment by jsancio** [comment]
Thanks for the review @cruhland. I will merge once it goes green again.

**Pull request merged**